pyo3-macros-backend/src/pyclass.rs

```diff
@@ -191,11 +195,7 @@ pub fn build_py_class(
     methods_type: PyClassMethodsType,
 ) -> syn::Result<TokenStream> {
     args.options.take_pyo3_options(&mut class.attrs)?;
-    let text_signature_string = text_signature_or_none(args.options.text_signature.as_ref());
```

Comment:
Sorry to answer myself, but this seems to have moved to `fn new`. Does this imply that the text processing happens at runtime now?

Reply:
There's a test to make sure that if `#[pyclass(text_signature = "...")]` is set, it's still used in preference to anything from `#[new]`, for backwards compatibility. Updated the PR description to explain this, sorry for lacking detail.

Yes, the text processing now necessarily happens at runtime for `#[new]. The `#[pyclass(name = "...")]` is only known by `#[pyclass]`, similarly for the doc. However the text_signature will now be coming from `#[pymethods]` (using dtolnay specialization).

Reply:
One thing I do not yet understand is why the specialization is required. Can't we just always generate an appropriate trait impl for each `#[pyclass]`? Is this for painlessly handling classes with out a `__new__` method at all?

Reply:
Yes, I think it's a valid use case to make a pyclass which can't be constructed from Python, so that's why I use the specialization to permit the option.